starter-code/src/clue.js

```diff
@@ -1,94 +1,184 @@
+age:          36,
+image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+occupation:   "Socialité"
+};
+
+var mrMustard = {
+first_name:   "Jack",
+last_name:    "Mustard",
+color:        "yellow",
+description:  "He is a former football player who tries to get by on his former glory",
+age:          62,
+image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+occupation:   "Retired Football player"
+};
 // Weapons
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var rope = {
+  name: "rope",
+  weight: 10
+};
+var knife = {
+  name: "knife",
+  weight: 8
+};
+var candlestick = {
+  name: "candlestick",
+  weight: 2
+};
+var dumbbell = {
+  name: "dumbbell",
+  weight: 30
+};
+var poison = {
+  name: "poison",
+  weight: 2
+};
+var axe = {
+  name: "axe",
+  weight: 15
+};
+var bat = {
+  name:"bat",
+  weight: 13
+};
+var trophy = {
+  name:"trophy",
+  weight: 25
+};
+var pistol = {
+  name:"pistol",
+  weight: 20
+};
+// Rooms' Collection
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var dinningRoom = { 
+  name: 'DinningRoom' 
+};
+var conservatory = { 
+  name: 'Conservatory'
+};
+var kitchen = { 
+  name: 'Kitchen'
+};
+var study = {
+  name: 'Study'
+};
+var library = {
+  name: 'Library'
+};
+var billiardRoom = {
+  name: 'BilliardRoom'
+};
+var lounge = {
+  name: 'Lounge'
+};
+var ballroom = {
+  name: 'Ballroom'
+};
+var hall = {
+  name: 'Hall'
+};
+var spa = {
+  name: 'Spa'
+};
+var livingRoom = {
+  name: 'LivingRoom'
+};
+var observatory = {
+  name: 'Observatory'
+};
+var theater = {
+  name: 'Theater'
+};
+var guestHouse = {
+  name: 'GuestHouse'
+};
+var patio = {
+  name: 'Patio'
+};
+
 
 // Characters Collection
 var charactersArray = [];
+charactersArray.push(mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard);
+
+// Weapons Collection
+var weaponsArray = [];
+weaponsArray.push(rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol);
+
 
 // Rooms' Collection
 var roomsArray = [];
+roomsArray.push(dinningRoom, conservatory, patio, theater, guestHouse, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory);
 
-// Weapons Collection
-var weaponsArray = [];
+var misteryEnvelope;
+
+function randomSelector(array){
+  var randomArray = array[Math.floor(Math.random()*array.length)];
+  return randomArray;
+}
+
+function pickMistery(array){
+  misteryEnvelope = [];
+  for (var i = 0; i < 1; i++){
```

Comment:
Por qué haces un bucle `for` aquí? No tiene mucho sentido porque solo das una vuelta. Además sabes que siempre darás solo una. Mejor devuelve la creación de un array con los elementos dentro: `return [randomSelector(charArr), randomSelector(weaponsArr), randomSelector(roomsArr)]`